pkg/controllers/federatedcluster/clusterstatus.go

```diff
@@ -60,8 +67,166 @@ const (
+		return
+	}
+	if len(pluginMaps) == 0 {
+		c.removeExternalClusterResourceInformers(cluster.Name)
+		return
+	}
+	pluginsHash := plugins.PluginsHash(pluginMaps)
+
+	c.lock.Lock()
+	defer c.lock.Unlock()
+
+	if old, exist := c.clusterConnectionHashes[cluster.Name]; exist {
+		// Connection and plugins are unchanged, do nothing
+		if old == clusterKey && c.enabledClusterResourcePluginHashes[cluster.Name] == pluginsHash {
+			return
+		}
+		// Otherwise, delete old informer since the connection has changed.
+		// We use the same context for a cluster, so if the plugins changed,
+		// we have to rebuild the informer.
+		if cancel := c.externalClusterResourceInformerCancelFuncs[old]; cancel != nil {
+			cancel()
+		}
+		delete(c.externalClusterResourceInformers, old)
+		delete(c.externalClusterResourceInformerCancelFuncs, old)
+	}
+
+	client, ok := c.federatedInformerManager.GetClusterDynamicClient(cluster.Name)
+	if !ok {
+		return
+	}
+	informer := dynamicinformer.NewDynamicSharedInformerFactory(client, 0)
+	ctx, cancel := context.WithCancel(context.Background())
+	for _, plugin := range pluginMaps {
+		for gvr := range plugin.ClusterResourcesToCollect() {
+			informer.ForResource(gvr)
+		}
+	}
+	informer.Start(ctx.Done())
+
+	c.clusterConnectionHashes[cluster.Name] = clusterKey
+	c.enabledClusterResourcePluginHashes[cluster.Name] = pluginsHash
+	c.externalClusterResourceInformers[clusterKey] = informer
+	c.externalClusterResourceInformerCancelFuncs[clusterKey] = cancel
+}
+
+func (c *FederatedClusterController) removeExternalClusterResourceInformers(clusterName string) {
+	c.lock.Lock()
+	defer c.lock.Unlock()
+
+	key := c.clusterConnectionHashes[clusterName]
+	if cancel := c.externalClusterResourceInformerCancelFuncs[key]; cancel != nil {
+		cancel()
+	}
+
+	delete(c.clusterConnectionHashes, clusterName)
+	delete(c.enabledClusterResourcePluginHashes, clusterName)
+	delete(c.externalClusterResourceInformers, key)
+	delete(c.externalClusterResourceInformerCancelFuncs, key)
+}
+
+func collectExternalClusterResources(
+	ctx context.Context,
+	availableNodes []*corev1.Node,
+	pods []*corev1.Pod,
+	resourceCollectors []resourceCollector,
+) {
+	wg := sync.WaitGroup{}
+	for i, collector := range resourceCollectors {
+		wg.Add(1)
+		go func(i int, collector resourceCollector) {
+			defer wg.Done()
+
+			ctx, logger := logging.InjectLoggerValues(ctx, "plugin", collector.name)
+			ctxWithTimeout, cancel := context.WithTimeout(ctx, 5*time.Second)
+			defer cancel()
+
+			hasSynceds := make([]cache.InformerSynced, 0, len(collector.hasSynceds))
+			names := make([]string, 0, len(collector.hasSynceds))
+			for gvr, synced := range collector.hasSynceds {
+				hasSynceds = append(hasSynceds, synced)
+				names = append(names, gvr.String())
+			}
+			if !cache.WaitForCacheSync(ctxWithTimeout.Done(), hasSynceds...) {
+				logger.V(4).Info("Timeout waiting for informers sync",
+					"informers", strings.Join(names, ";"))
+				return
```

Comment:
Should we continue to collect the resource information for the gvrs that have managed to sync?

Reply:
Oh my bad, the collection is performed by the plugin and might depend on all the  gvrs being synced